lnwallet/errors.go

```diff
@@ -0,0 +1,104 @@
+package lnwallet
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcutil"
+)
+
+// Reservation error wraps certain errors returned during channel reservation
+// that can be sent across the wire to the remote peer. Errors not being
+// ReservationErrors will not be sent to the remote in case of a failed channel
+// reservation, as they may contain private information.
+type ReservationError struct {
+	error
```

Comment:
Why the embedding here? It's an interface, so all we need to do is ensure that the structs implement the `Error() string` method.

Reply:
So this can just be a `string` field minimally. Though in a way this _is_ less code by just embedding an error...

Reply:
I embedded the error such that they all can be checked whether they are of the type `ReservationError`, centralizing the list of "acceptable" reservation errors. It also embed the error messages containing non-constant arguments (which as you mention can just as easily be done with a `string` field).

Another way to do this would be to make it more like to the `routing` errors, where the `ReservationError` contains an error code maybe, but not sure if there's an easy way to have the dynamic arguments (minimum amounts etc) this way?